backend/TreeOfAKind.API/UserProfiles/UserProfileController.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using TreeOfAKind.API.SeedWork;
+using TreeOfAKind.Application.Command.UserProfiles;
+using TreeOfAKind.Application.Command.UserProfiles.CreateOrUpdateUserProfile;
+using TreeOfAKind.Application.Query.UserProfiles.GetMyUserProfile;
+using TreeOfAKind.Domain.UserProfiles;
+
+namespace TreeOfAKind.API.UserProfiles
+{
+    [Route("api/[controller]/[action]")]
+
+    public class UserProfileController : Controller
+    {
+        private readonly IMediator _mediator;
+
+        public UserProfileController(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        /// <summary>
+        /// Creates or updates (if it already exists) User Profile for authenticated user.
+        /// </summary>
+        /// <remarks>
+        /// Sample request:
+        ///
+        ///     POST /UserProfile/CreateUserProfile
+        ///     {
+        ///        "firstName": "Bartosz",
+        ///        "lastName": "Chrostowski",
+        ///        "birthDate": "1998-02-12T00:00:00.000Z"
+        ///     }
+        ///
+        /// </remarks>
+        /// <param name="request"></param>
+        /// <returns>Uuid of created user profile</returns>
+        /// <response code="201">Returns uuid of created user profile</response>
```

Comment:
That's not very command-like.

Reply:
![image](https://user-images.githubusercontent.com/32818862/98030978-1b3a9200-1e12-11eb-9508-d2a6b996945c.png)
[ling](https://vladikk.com/2017/03/20/tackling-complexity-in-cqrs/)

Reply:
Also, don't treat api as strictly cqrs compliant

I suppose there may be some non-cqrs endpoints in the future and nothing's bad with it